malt/converters/slices.py

```diff
@@ -52,6 +52,23 @@ def visit_Assign(self, node):
     if replacement is not None:
       return replacement
     return node
+  
+  def visit_Slice(self, node):
+    node = self.generic_visit(node)
+    if not isinstance(node, (gast.Slice)):
+      return node
+    # This guard is required or the following example will produce a wrong
+    # data shape.
+    #   @qjit(autograph=True)
+    #   def expand_by_two(x):
+    #     print(*.x.shape[1:])
+    if node.lower is None or node.upper is None:
+      return node
```

Comment:
Why does it produce the wrong shape? we should be able to convert slices with implicit start and end.

Reply:
``slice(n)`` is not the same as ``n:``.  ``slice(n)`` is equivalent to  ``slice(0, n)`` or ``:n``. I will fix that.